### DISCUSSION TITLE
wasm或者hybird在什么地方初始化载入设置默认文化信息?

### DISCUSSION THREAD

**Discussion by densen2014** [dotnetcore/BootstrapBlazor#3399] [Q&A] [answered]
wasm或者hybird在什么地方初始化载入设置默认文化信息? 我现在在 MainLayout 的 OnAfterRender 用 culture= Storage.GetValue 后 CultureInfo.DefaultThreadCurrentCulture = culture 会有什么问题吗

**Comment by ArgoZhang** [comment, marked as answer]
### MAUI 

MauiProgram file

```
public static MauiApp CreateMauiApp()
{
    // 增加 BootstrapBlazor 组件服务
    builder.Services.AddBootstrapBlazor();

    // 增加 Table 数据服务操作类
    builder.Services.AddTableDemoDataService();

    CultureInfo.DefaultThreadCurrentUICulture = new CultureInfo("en-US");

    return builder.Build();
}
```

Build 或者 Run 之前设置下面即可

`CultureInfo.DefaultThreadCurrentUICulture = new CultureInfo("en-US");`

**Reply by densen2014** [reply]
可是我要的是读存在浏览器LocalStorage里面的变量, 应该怎样才合理呢, wasm工程取消了我没法研究了

**Reply by ArgoZhang** [reply]
那就麻烦点，用户体验好一点的做法是

1. 首次渲染什么都不加载（加快速度）直接到 `OnAfterRenderAsync`
2. `OnAfterRenderAsync` 中读取 `LocalStorage` 中的设置
3. 根据读取到的设置更改文化信息，再 `StateHasChanged`
